crates/ruff_linter/src/rules/pylint/rules/too_many_attributes.rs

```diff
@@ -0,0 +1,113 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{identifier::Identifier, Expr, ExprAttribute, Stmt, StmtClassDef};
+use ruff_python_semantic::{Scope, ScopeKind};
+use std::collections::HashSet;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for classes that include too many class attributes.
+///
+/// By default, this rule allows up to seven attributes, as configured by the
+/// [`lint.pylint.max-attributes`] option.
+///
+/// ## Why is this bad?
+/// Classes with many attributes are harder to use and maintain.
+///
+/// Consider reducing class attributes to get a simpler and easier to use class.
+///
+/// ## Options
+/// - `lint.pylint.max-attributes`
+#[violation]
+pub struct TooManyAttributes {
+    current_amount: usize,
+    max_amount: usize,
+}
+
+impl Violation for TooManyAttributes {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooManyAttributes {
+            current_amount,
+            max_amount,
+        } = self;
+        format!("Too many instance attributes ({current_amount}/{max_amount})")
+    }
+}
+
+fn collect_attributes<'a>(attributes: &mut HashSet<&'a str>, expr: &'a ExprAttribute) {
+    if let Some(ident) = expr.value.as_name_expr() {
+        if ident.id == "self" {
+            attributes.insert(expr.attr.as_str());
+        }
+    }
+}
+
+fn extract_assigned_attributes<'a>(attributes: &mut HashSet<&'a str>, target: &'a Expr) {
+    match target {
+        Expr::Attribute(expr) => {
+            collect_attributes(attributes, expr);
+        }
+        Expr::Tuple(tuple) => {
+            for expr in &tuple.elts {
+                if let Some(expr) = expr.as_attribute_expr() {
+                    collect_attributes(attributes, expr);
+                }
+            }
+        }
+        _ => (),
+    }
+}
+
+/// PLR0902
+pub(crate) fn too_many_attributes(
+    checker: &Checker,
+    scope: &Scope,
```

Comment:
I suspect that we don't need the `scope` here, because it's given that we're inside of a class, because we get the `class_def`

Reply:
Addressed.